rskj-core/src/main/java/co/rsk/net/light/LightSyncProcessor.java

```diff
@@ -0,0 +1,190 @@
+                boolean shouldDiscard = size() > MAX_PENDING_MESSAGES;
+                if (shouldDiscard) {
+                    loggerNet.trace("Pending {}@{} DISCARDED", eldest.getValue(), eldest.getKey());
+                }
+                return shouldDiscard;
+            }
+        };
+    }
+
+    public void processStatusMessage(StatusMessage msg, LightPeer lightPeer, ChannelHandlerContext ctx, LightClientHandler lightClientHandler) {
+        LightStatus status = msg.getStatus();
+        try {
+            String bestHashLog = HashUtil.shortHash(status.getBestHash());
+            loggerNet.debug("Receiving Status - block {} {}", status.getBestNumber(), bestHashLog);
+
+            byte protocolVersion = status.getProtocolVersion();
+            if (protocolVersion != version) {
+                loggerNet.info("Removing LCHandler for {} due to protocol incompatibility", ctx.channel().remoteAddress());
+                loggerNet.info("Protocol version {} - message protocol version {}",
+                        version,
+                        protocolVersion);
+                lightPeer.disconnect(ReasonCode.INCOMPATIBLE_PROTOCOL);
+                ctx.pipeline().remove(lightClientHandler); // Peer is not compatible for the 'lc' sub-protocol
+                return;
+            }
+
+            int networkId = config.networkId();
+            int msgNetworkId = status.getNetworkId();
+            if (msgNetworkId != networkId) {
+                loggerNet.info("Removing LCHandler for {} due to invalid network", ctx.channel().remoteAddress());
+                loggerNet.info("Different network received: config network ID {} - message network ID {}",
+                        networkId, msgNetworkId);
+                lightPeer.disconnect(ReasonCode.NULL_IDENTITY);
+                ctx.pipeline().remove(lightClientHandler);
+                return;
+            }
+
+            Keccak256 genesisHash = genesis.getHash();
+            Keccak256 msgGenesisHash = new Keccak256(status.getGenesisHash());
+            if (!msgGenesisHash.equals(genesisHash)) {
+                loggerNet.info("Removing LCHandler for {} due to unexpected genesis", ctx.channel().remoteAddress());
+                loggerNet.info("Config genesis hash {} - message genesis hash {}",
+                        genesisHash, msgGenesisHash);
+                lightPeer.disconnect(ReasonCode.UNEXPECTED_GENESIS);
+                ctx.pipeline().remove(lightClientHandler);
+                return;
+            }
+        } catch (NoSuchElementException e) {
+            loggerNet.debug("LCHandler already removed - exception: {}", e.getMessage());
+        }
+
+        if (peerStatuses.size() >= MAX_PEER_CONNECTIONS) {
+            return;
+        }
+
+        if (!hasLowerDifficulty(status)) {
+            return;
+        }
+
+        peerStatuses.put(lightPeer, status);
+
+        if (msg.isTxRelay()) {
```

Comment:
This is always false, except on the unit tests. Right?

Reply:
It's false by default